lib/teleterm/apiserver/apiserver.go

```diff
@@ -66,24 +79,30 @@ func (s *APIServer) Stop() {
 }
 
 func newListener(hostAddr string) (net.Listener, error) {
-	uri, err := url.Parse(hostAddr)
+	uri, err := utils.ParseAddr(hostAddr)
 
 	if err != nil {
 		return nil, trace.BadParameter("invalid host address: %s", hostAddr)
 	}
 
-	if uri.Scheme != "unix" {
-		return nil, trace.BadParameter("invalid unix socket address: %s", hostAddr)
-	}
-
-	lis, err := net.Listen(uri.Scheme, uri.Path)
+	lis, err := net.Listen(uri.Network(), uri.Addr)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	addr := utils.FromAddr(lis.Addr())
+	sendBoundNetworkPortToStdout(addr)
+
+	log.Infof("tsh daemon is listening on %v.", addr.FullAddress())
+
 	return lis, nil
 }
 
+func sendBoundNetworkPortToStdout(addr utils.NetAddr) {
+	// The Electron app needs this message to know which port has been assigned to the server. 
+	fmt.Printf("{CONNECT_GRPC_PORT: %v}\n", addr.Port(1))
```

Comment:
If this is meant to be json, shouldn't it have quotes around it?
```suggestion
	fmt.Printf("{\"CONNECT_GRPC_PORT\": %v}\n", addr.Port(1))
```

Reply:
Also, now that I think about it, why can't we also send our certificate to Connect in the same way, and receive it as a base64-encoded parameter or something along those lines?

Reply:
>If this is meant to be json, shouldn't it have quotes around it?

Oops, coincidence, I didn't mean to create JSON.
 
>Also, now that I think about it, why can't we also send our certificate to Connect in the same way, and receive it as a base64-encoded parameter or something along those lines?

Yeah, it could be done this way, but the current solution nicely encapsulates certs reading & generation in `grpcCredentials`, so only the place that creates the client or server has to use that function (also the logic on both sides is the same). If we would want to send a cert with process params we would have to generate it before starting the process (and we are going back to the first iteration 😅). I'd stay with what we have now, it seems to be the cleanest way.